client/platform/desktop/backend/native/common.ts

```diff
@@ -0,0 +1,473 @@
+ *
+ * SUPPORTED FORMATS:
+ * VIAME CSV
+ *
+ * @param paths paths to possible input annotation files
+ * @param datasetId dataset id path
+ */
+async function processOtherAnnotationFiles(
+  settings: Settings,
+  datasetId: string,
+  absPaths: string[],
+): Promise<{ fps?: number; processedFiles: string[] }> {
+  const fps = undefined;
+  const processedFiles = []; // which files were processed to generate the detections
+
+  for (let i = 0; i < absPaths.length; i += 1) {
+    const path = absPaths[i];
+    if (!fs.existsSync(path)) {
+      // eslint-disable-next-line no-continue
+      continue;
+    }
+    if (fs.statSync(path).size > 0) {
+      // Attempt to process the file
+      // eslint-disable-next-line no-await-in-loop
+      const tracks = await viameSerializers.parseFile(path);
+      const data: MultiTrackRecord = {};
+      tracks.forEach((t) => { data[t.trackId.toString()] = t; });
+      // eslint-disable-next-line no-await-in-loop
+      await _saveSerialized(settings, datasetId, data);
+      processedFiles.push(path);
+      break; // Exit on first successful detection load
+    }
+  }
+  return { fps, processedFiles };
+}
+
+async function _initializeAppDataDir(settings: Settings) {
+  await fs.ensureDir(settings.dataPath);
+  await fs.ensureDir(npath.join(settings.dataPath, ProjectsFolderName));
+  await fs.ensureDir(npath.join(settings.dataPath, JobsFolderName));
+}
+
+/**
+ * Intialize a new project directory
+ * @returns absolute path to new project dcirectory
+ */
+async function _initializeProjectDir(settings: Settings, jsonMeta: JsonMeta): Promise<string> {
+  const projectDir = npath.join(settings.dataPath, ProjectsFolderName, jsonMeta.id);
+  await _initializeAppDataDir(settings);
+  await fs.ensureDir(projectDir);
+  return projectDir;
+}
+
+/**
+ * importMedia locates as much information as possible
+ * about a dataset using only the directory structure.
+ * @param settings user settings
+ * @param path path to import dir/file
+ * @returns datasetId
+ */
+async function importMedia(settings: Settings, path: string): Promise<JsonMeta> {
+  let datasetType: DatasetType;
+
+  const exists = fs.existsSync(path);
+  if (!exists) {
+    throw new Error(`file or directory not found: ${path}`);
+  }
+
+  const stat = await fs.stat(path);
+  if (stat.isDirectory()) {
+    datasetType = 'image-sequence';
+  } else if (stat.isFile()) {
+    datasetType = 'video';
+  } else {
+    throw new Error('Only regular files and directories are supported');
+  }
+
+  const dsName = npath.parse(path).name;
+  const dsId = `${cleanString(dsName).substr(0, 20)}_${makeid(10)}`;
+
+  const jsonMeta: JsonMeta = {
+    version: JsonMetaCurrentVersion,
+    type: datasetType,
+    id: dsId,
+    fps: 5, // TODO
+    originalBasePath: path,
+    originalVideoFile: '',
+    originalImageFiles: [],
+    transcodedVideoFile: '', // TODO: this is empty (see above)
+    transcodedImageFiles: [], // TODO: this is empty
+    name: dsName,
+  };
```

Comment:
This may be a niche case, but we may want to add either a timestamp or a user custom Name to the JSONMeta.  Feel free to dismiss or mark it for another PR - During testing I opened the same dataset multiple times, it properly creates the different unique directories and everything, but then I can't tell the difference between them and some have pipelines run on them, some have other input data that I used.  Just wondering if in metadata we should have either a timestamp or something else to let the user know more information about dataset or a user configurable name?
![image](https://user-images.githubusercontent.com/61746913/103413534-047d9680-4b48-11eb-821f-f821f02cc25a.png)

Reply:
Good suggestion, we should maintain a `createdAt` property in metadata.  I don't know if we should show it in recents because it may look messy.  But it would be good to have.

Reply:
Only added to metadata, didn't render it in the resents list.  It looks messy and wraps badly.